app/attachments.js

```diff
@@ -167,3 +169,63 @@ exports.createAbsolutePathGetter = rootPath => relativePath => {
   }
   return normalized;
 };
+
+//      getTempPath :: AbsolutePath -> AbsolutePath
+exports.getTempPath = userDataPath => {
+  if (!isString(userDataPath)) {
+    throw new TypeError("'userDataPath' must be a string");
+  }
+  return path.join(userDataPath, TEMPPATH);
+};
+
+//      createWriterForTempAttachment :: AttachmentsPath -> ArrayBuffer -> Ext ->
+//                            { Filename , ArrayBuffer }
+exports.createWriterForTempAttachment = root => {
+  if (!isString(root)) {
+    throw new TypeError("'root' must be a path");
+  }
+
+  return async (arrayBuffer) => {
+    fse.ensureDirSync(root);
+
+    const buffer = Buffer.from( new Uint8Array(arrayBuffer) );
+    const filename = `${uuidv4()}`;
+    const tempFilePath = path.join(root, filename);
+
+    fse.writeFileSync(tempFilePath, buffer);
+
+    const newBuffer = fse.readFileSync(tempFilePath);
```

Comment:
This and the next five lines - what problem is it trying to solve? It's not immediately clear that it's necessary, so at the very least it needs a comment.